subprojects/packagefiles/tree-sitter/meson.build

```diff
@@ -0,0 +1,17 @@
+project('tree-sitter', 'c')
+
+tree_sitter_lib = static_library('tree-sitter-runtime', [
+  'lib/src/lib.c',
+  ]
```

Comment:
Upstream's Makefile creates and installs this as 'tree-sitter', and generates both static and shared libraries, so you should not hardcode it as static but rather allow the user to select it via `-Ddefault_library=`.

lib/src/lib.c is an amalgamated file that is updated less frequently than lib/src/*.c, I haven't checked when/how often it gets updated. :/ Not sure whether that matters. The Makefile only uses the amalgamation with `make AMALGAMATED=1`.

This should have an install target, including installing lib/include/tree_sitter/ to includedir, and generate a pkg-config file.

Reply:
ok, makes sense, thanks. Working on it

Reply:
The install target means also `install: true` on the library.

Reply:
apart from specify all detailed sources I think this is done

Reply:
Upstream also has a `soname: '0.0'` equivalent in the Makefile.